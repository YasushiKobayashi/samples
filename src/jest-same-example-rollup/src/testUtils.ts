const testUtils = () => {
  return 'test utils method'
}

export default testUtils
