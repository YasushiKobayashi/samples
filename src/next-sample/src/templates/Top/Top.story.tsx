import * as React from 'react'
import { expect, jest } from '@storybook/jest'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import { userEvent, within } from '@storybook/testing-library'
import { waitFor } from '@testing-library/react'

import { Top } from './Top'

export default {
  title: 'templates/Top',
  component: Top,
} as ComponentMeta<typeof Top>

const base = {
  submit: jest.fn(),
}

const PrimaryTemplate: ComponentStory<typeof Top> = additionalProps => {
  base.submit.mockClear()
  const props = { ...base, ...additionalProps }
  return <Top {...props} />
}
export const Primary = PrimaryTemplate.bind({})
Primary.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  await waitFor(async () => {
    await userEvent.type(canvas.getByLabelText('First Name'), 'LeBron')
    await userEvent.type(canvas.getByLabelText('Last Name'), 'James')
    await userEvent.click(canvas.getByRole('button', { name: 'submit' }))
  })
  expect(base.submit).toHaveBeenCalledWith({
    firstName: 'LeBron',
    lastName: 'James',
  })
  expect(base.submit).toHaveBeenCalledTimes(1)
}
