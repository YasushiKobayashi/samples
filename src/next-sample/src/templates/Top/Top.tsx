import * as React from 'react'

import { InputForm } from '@/atoms/InputForm/InputForm'

interface Form {
  firstName: string
  lastName: string
}

const initialState: Form = {
  firstName: '',
  lastName: '',
}

interface Props {
  submit: (val: Form) => void
}

export const Top: React.FC<Props> = ({ submit }) => {
  const [form, setForm] = React.useState<Form>({ ...initialState })

  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    submit(form)
  }

  return (
    <form onSubmit={onSubmit}>
      <div style={{ marginBottom: 16 }}>
        <InputForm
          val={form.firstName}
          id="firstName"
          label="First Name"
          onChange={v => setForm({ ...form, firstName: v })}
        />
      </div>
      <div style={{ marginBottom: 16 }}>
        <InputForm
          val={form.lastName}
          id="lastName"
          label="Last Name"
          onChange={v => setForm({ ...form, lastName: v })}
        />
      </div>
      <button type="submit">submit</button>
    </form>
  )
}
