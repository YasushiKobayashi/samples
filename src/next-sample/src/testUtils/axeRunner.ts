import { configureAxe } from 'jest-axe'

export const axeRunner = configureAxe({
  rules: {},
})
