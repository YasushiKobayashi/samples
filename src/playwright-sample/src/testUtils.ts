import AxeBuilder from '@axe-core/playwright'

export const baseUrl = 'http://localhost:3000/'
export const baseDir = 'test-results/screenshots'

const axeRunner = async (page: Page, disableRules: string[] = []) => {
  const results = await new AxeBuilder({ page })
    .disableRules(['image-alt', 'color-contrast', 'meta-viewport', 'link-name', ...disableRules])
    .analyze()

  if (results.violations.length > 0) {
    const title = await page.title()
    const url = page.url()
    console.error(title, url, results.violations)
    // test.fail()
  }
}

let i = 1
export const testPage = async (page: Page, basePath: string, disableRules: string[] = []) => {
  await axeRunner(page, disableRules)
  // eslint-disable-next-line no-plusplus
  await page.screenshot({ path: `${basePath}/${i++}.png`, fullPage: true })
}
