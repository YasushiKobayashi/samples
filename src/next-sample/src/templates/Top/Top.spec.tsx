import * as React from 'react'
import { composeStories } from '@storybook/react'
import { act, cleanup, render } from '@testing-library/react'

import { axeRunner } from '@/testUtils/axeRunner'

import * as stories from './Top.story'

const { Primary } = composeStories(stories)

describe('templates/Top', () => {
  afterEach(() => {
    cleanup()
  })

  it('Snap Shot', async () => {
    const { container, asFragment } = render(<Primary />)
    expect(asFragment()).toMatchSnapshot()

    await act(async () => {
      if (Primary.play) Primary.play({ canvasElement: container })
      expect(await axeRunner(container)).toHaveNoViolations()
    })
  })
})
