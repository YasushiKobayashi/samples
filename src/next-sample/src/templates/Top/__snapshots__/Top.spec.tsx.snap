// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates/Top Snap Shot 1`] = `
<DocumentFragment>
  <form>
    <div
      style="margin-bottom: 16px;"
    >
      <div>
        <label
          for="firstName"
        >
          First Name
        </label>
        <input
          id="firstName"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      style="margin-bottom: 16px;"
    >
      <div>
        <label
          for="lastName"
        >
          Last Name
        </label>
        <input
          id="lastName"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</DocumentFragment>
`;
